src/interface/src/app/password-reset/password-reset.component.html

```diff
@@ -1,52 +1,90 @@
+            <input
+              type="password"
+              required
+              formControlName="password1"
+              (focus)="showHint = true"
+              (blur)="showHint = false"
+              matInput />
+          </mat-form-field>
+          <app-field-alert
+            [messageType]="FormMessageType.ALERT"
+            *ngIf="showHint && !form.get('password1')?.value"
+            title="Password requirements"
+            message="Your new password needs to be at least 8 characters long and must not have been previously used.">
+          </app-field-alert>
 
-    <form [formGroup]="form" (ngSubmit)="submit()" class="email-form">
-      <mat-form-field appearance="fill">
-        <mat-label>Create password</mat-label>
-        <input type="password" required formControlName="password1" matInput />
-        <mat-error *ngIf="form.get('password1')?.hasError('minlength')">
-          Password must contain at least 8 characters.
-        </mat-error>
-      </mat-form-field>
+          <app-field-alert
+            *ngIf="
+              form.get('password1')?.hasError('minlength') &&
+              form.get('password1')?.touched
+            "
+            title="Field error"
+            message="Password must contain at least 8 characters.">
+          </app-field-alert>
 
-      <mat-form-field appearance="fill">
-        <mat-label>Confirm password</mat-label>
-        <input type="password" required formControlName="password2" matInput />
-        <mat-error *ngIf="form.hasError('passwordsNotEqual')">
-          Passwords must match.
-        </mat-error>
-      </mat-form-field>
+          <mat-label class="standard-label"
+            >Confirm Password
+            <span class="required-blurb">required</span>
+          </mat-label>
+          <mat-form-field appearance="outline">
+            <input
+              type="password"
+              required
+              formControlName="password2"
+              matInput
+              autocomplete="current-password"
+              [errorStateMatcher]="currentPasswordStateMatcher" />
+          </mat-form-field>
+          <app-field-alert
+            [messageType]="FormMessageType.ERROR"
+            *ngIf="form.errors?.['newPasswordsMustMatch']"
+            title="Password Error"
+            message="Both entries need to match">
+          </app-field-alert>
 
-      <div class="reset-form-buttons">
-        <button
-          mat-stroked-button
-          color="secondary"
-          type="button"
-          class="reset-form-button"
-          (click)="cancel()">
-          Cancel
-        </button>
-        <button
-          mat-flat-button
-          color="primary"
-          type="submit"
-          [disabled]="!form.valid"
-          class="reset-form-button">
-          Reset Password
-        </button>
-      </div>
+          <app-form-message
+            [messageType]="FormMessageType.ERROR"
+            *ngIf="errors.length > 0"
```

Comment:
do you have `errors`? should this be `form.errors` instead?

Reply:
Yeah, I think I was trying to collect just backend errors separate from the form.errors, which was a bad idea. Changed this pattern.